test/metabase/lib/filter_test.cljc

```diff
@@ -372,16 +372,38 @@
               :database-type "BOOLEAN",
               :effective-type :type/Boolean,
               :fk-target-field-id nil,
-              :operators [{:lib/type :operator/filter, :short :=, :display-name-variant :default}
-                          {:lib/type :operator/filter, :short :is-null, :display-name-variant :is-empty}
-                          {:lib/type :operator/filter, :short :not-null, :display-name-variant :not-empty}],
               :id 14,
               :parent-id nil,
               :visibility-type :normal,
               :lib/desired-column-alias "TRIAL_CONVERTED",
               :display-name "Trial Converted",
               :position 10,
-              :fingerprint {:global {:distinct-count 2, :nil% 0.0}}})))))
+              :fingerprint {:global {:distinct-count 2, :nil% 0.0}}}))))
+  (testing "should return text-like operators for text-like PKs and FKs"
+    (for [semantic-type [:type/PK :type/FK]
+          :let [column {:description nil,
+                        :lib/type :metadata/column,
+                        :base-type :type/MongoBSONID,
+                        :semantic-type semantic-type
+                        :table-id 7,
+                        :name "ID",
+                        :coercion-strategy nil,
+                        :lib/source :source/table-defaults,
+                        :lib/source-column-alias "ID",
+                        :settings nil,
+                        :lib/source-uuid "ad9a276f-3af8-4e5a-b17e-d8170273ec0a",
+                        :nfc-path nil,
+                        :database-type "STRING",
+                        :effective-type :type/MongoBSONID,
+                        :fk-target-field-id nil,
+                        :id 14,
+                        :parent-id nil,
+                        :visibility-type :normal,
+                        :lib/desired-column-alias "ID",
+                        :display-name "ID",
+                        :position 10}]]
+        (is (= [:= :!= :is-null :not-null :is-empty :not-empty]
+               (mapv :short (lib.filter.operator/filter-operators column)))))))
```

Comment:
```suggestion
    (doseq [semantic-type [:type/PK :type/FK]
            :let [column {:description nil,
                          :lib/type :metadata/column,
                          :base-type :type/MongoBSONID,
                          :semantic-type semantic-type
                          :table-id 7,
                          :name "ID",
                          :coercion-strategy nil,
                          :lib/source :source/table-defaults,
                          :lib/source-column-alias "ID",
                          :settings nil,
                          :lib/source-uuid "ad9a276f-3af8-4e5a-b17e-d8170273ec0a",
                          :nfc-path nil,
                          :database-type "STRING",
                          :effective-type :type/MongoBSONID,
                          :fk-target-field-id nil,
                          :id 14,
                          :parent-id nil,
                          :visibility-type :normal,
                          :lib/desired-column-alias "ID",
                          :display-name "ID",
                          :position 10}]]
        (is (= [:= :!= :is-null :not-null :is-empty :not-empty]
               (mapv :short (lib.filter.operator/filter-operators column)))))))
```

Reply:
I suggest using `doseq`. `for` returns lazy sequence that has to be consumed for side effects to take place. On the other hand, `doseq` is meant exactly for that -- realizing the sequence for effects, while returning `nil`.

You could verify what I wrote executing test swapping the suggestion for following snippet:
```
(for [semantic-type [:type/PK :type/FK]
          :let [column {:description nil,
                        :lib/type :metadata/column,
                        :base-type :type/MongoBSONID,
                        :semantic-type semantic-type
                        :table-id 7,
                        :name "ID",
                        :coercion-strategy nil,
                        :lib/source :source/table-defaults,
                        :lib/source-column-alias "ID",
                        :settings nil,
                        :lib/source-uuid "ad9a276f-3af8-4e5a-b17e-d8170273ec0a",
                        :nfc-path nil,
                        :database-type "STRING",
                        :effective-type :type/MongoBSONID,
                        :fk-target-field-id nil,
                        :id 14,
                        :parent-id nil,
                        :visibility-type :normal,
                        :lib/desired-column-alias "ID",
                        :display-name "ID",
                        :position 10}]]
      (do (println semantic-type)
          (is (= [:= :!= :is-null :not-null :is-empty :not-empty]
                 (mapv :short (lib.filter.operator/filter-operators column))))))))
```